### ISSUE TITLE
VCMI 1.5.6 crashing way too often

### ISSUE THREAD

**Issue by Raikon117** [vcmi/vcmi#4441]
**Describe the bug**
VCMI is crashing constantly and absolutely impossible to play. Before combat, at end of turns etc. 
In addition some moduls cannot be loaded because it's saying that are incompatible with others.
Attaching logs.

**Game logs**
[VCMI_Client_log.txt](https://github.com/user-attachments/files/16588195/VCMI_Client_log.txt)
[VCMI_Launcher_log.txt](https://github.com/user-attachments/files/16588196/VCMI_Launcher_log.txt)

**To Reproduce**

**Expected behavior**
To run correctly.

**Actual behavior**
Crashing like hell

**Did it work earlier?**
N/A

**Screenshots**
I can if it's necessary

**Version**
 -  Windows 10 
 - Version: [VCMI version]

**Additional context**
N/A

**Comment by dydzio0614** [comment]
can you test this build and tell if it improves anything? https://github.com/vcmi/vcmi/releases/download/1.5.6/VCMI-WindowsAlternate.exe

**Comment by Raikon117** [comment]
Now it's not starting at all. It gives me error directly when I start a map.

**Comment by dasassaa** [comment]
Yeah, it is almost unplayable right now. Playing a single map, VCMI can crash up to 30 times until one can finish the map. I hope it will be fixed with the next release.

Maybe it's the same issue I reported with #4430

**Comment by kambala-decapitator** [comment]
interestingly that I had zero crashes on macOS so far. I use only Boost AI mod and play only random maps.

**Comment by IvanSavenko** [comment]
> spell id -1 is invalid

Most likely issue with some mod, but unfortunately can't find out which one. Could be related to `farriery-town`

**Comment by Raikon117** [comment]
You don't have a clue from the logs ?

**Comment by misiokles** [comment]
Disable farriery-town mod and check if crashes occur too (with new scenario, of course)

**Comment by Raikon117** [comment]
Another feedback : Death Valey town is crashing on all maps templates. Still didn't test all the towns, but Death Valey crashed on 4 different templates on a row on starting.

**Comment by misiokles** [comment]
Can you try to update Death Valley mod to see if crashes still occur?
And I can't see nothing suspicious in Farriery mod :(

**Comment by Raikon117** [comment]
It's not Farriery, but Fairy town I removed. 
I will test all towns and give feedback later

**Comment by misiokles** [comment]
Connectors are part of newtown terrains mod.

**Comment by Raikon117** [comment]
OK I did much more tests and unfortunately it's full of issues and bugs. A lot of maps are crashing with most of the towns and some towns are crashing all the time, no matter template. 
I think that it should be reworked and be way more stable. 
Anyway it's very good as idea and towns, musics, heroes. Me personally I would like to have option for experience to the units. And some of the new towns which are based on Castle to be different, but this is not the most important. 
Also to have option when taking scholars or tents to choose if we want to take the skill or not.

**Comment by misiokles** [comment]
Most likely broken mod (Crossroads of Worlds? An's Expansion?). Now is ongoing pvp tournament and players don't report anything like that. Please paste screen with list of your enebled mods.

**Comment by FitzRoyX** [comment]
I just started playing and got to the third map (Retrieving the Cowl) in the Shadow campaign which started crashing a ton for me once the enemy tried to seige me. Crashes would happen after ending turns, sometimes right after saving which would trap me into a crash. I was on 1.5.7 Windows with no mods enabled except for the vcmi extras. I tried upgrading to latest dev build, still crashed. I tried changing out the ai settings, still crashed. I tried restarting the map, still crashed. Don't have my saves any more but I gave up and went over to another engine. I really think you guys should focus on stability and basic qol.

**Comment by Olegmods** [comment]
> I just started playing and got to the third map (Retrieving the Cowl) in the Shadow campaign which started crashing a ton for me once the enemy tried to seige me. Crashes would happen after ending turns, sometimes right after saving which would trap me into a crash. I was on 1.5.7 Windows with no mods enabled except for the vcmi extras. I tried upgrading to latest dev build, still crashed. I tried changing out the ai settings, still crashed. I tried restarting the map, still crashed. Don't have my saves any more but I gave up and went over to another engine. I really think you guys should focus on stability and basic qol.

Did you try the alternate build?

**Comment by Olegmods** [comment]
> I really think you guys should focus on stability and basic qol.

Also I guess, crashlog could be helpful to found your issue. Also it's likely to be fixed in beta, which has more commits than the develop.

**Comment by FitzRoyX** [comment]
> Did you try the alternate build?

Sorry, I'm a newcomer, I can't really know about all the dev branches and forks with crash fixes. I did not try the "beta" branch, but I did try the latest "develop" action build and a fork from a guy who claimed to fix the crash issues in a different thread (xilmo?).

**Comment by Olegmods** [comment]
> > Did you try the alternate build?
> 
> Sorry, I'm a newcomer, I can't really know about all the dev branches and forks with crash fixes. I did not try the "beta" branch, but I did try the latest "develop" action build and a fork from a guy who claimed to fix the crash issues in a different thread (xilmo?).

Xilmi only fixed the AI. Try that build: https://builds.vcmi.download/branch/beta/Windows/VCMI-branch-beta-bfbc97c.exe , maybe that'll fix the issues you've encountered.
